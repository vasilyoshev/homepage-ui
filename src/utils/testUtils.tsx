import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

// TODO extract mocks to separate files, e.g. widgetsMock.js
const mockState = {
  loading: {},
  widgets: { widgetsProps: [] },
  user: {},
};
const createMockStore = configureStore([thunk]);
const store = createMockStore(mockState);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Providers = ({ children }: any) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
  </Provider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
