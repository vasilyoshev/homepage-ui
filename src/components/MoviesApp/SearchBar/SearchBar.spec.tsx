import { render } from 'utils';
import { SearchBar } from './SearchBar';

describe('SearchBar', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<SearchBar />);

    expect(asFragment()).toMatchSnapshot();
  });
});
