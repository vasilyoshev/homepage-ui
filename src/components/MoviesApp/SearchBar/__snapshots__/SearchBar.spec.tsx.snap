// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
        >
          Movies Collection
        </div>
        <div
          class="css-1orzy9s"
        >
          <div
            class="css-1iz4iln"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
